bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/json/SafeExposureTime.java

```diff
@@ -52,9 +52,14 @@ public State getSafeExposure(int index) {
             case 5:
                 result = st5;
                 break;
-            default:
+            case 6:
                 result = st6;
+                break;
+            default:
+                result = BigInteger.ZERO;
+
         }
-        return result != BigInteger.ZERO ? new QuantityType<>(result, SmartHomeUnits.MINUTE) : UnDefType.NULL;
+        return (result != BigInteger.ZERO && result != null) ? new QuantityType<>(result, SmartHomeUnits.MINUTE)
```

Comment:
if `result` is an int it can't be null. You can also use `0` instead of `BigInteger.ZERO` then. If you leave it at `Integer` the null-check is redunant, too. You definitely assign a value in l. 59.

Reply:
Yes, I know it's seems to be a bit overkill but for an unknown reason, I still get some NPE from time to time upon creation of QuantityType :-/

Reply:
The default should have been assigning st6 that is initialized to ZERO. no ?

Reply:
Do you mean that the private fields `st1` - `st6` are set by reflection using GSON? Maybe we are all better of, if you mark them nullable, because now the code does not make any sense. Maybe it would also make sense then to NOT initialise all the `st1` - `st6` and maybe make them public so that its less unexpected that those private fields change :-)

Reply:
Yes, you're right. Trying to circumvent it this way has become ununderstandable. Let's rewrite this.